specs/waku/waku-v2.md

```diff
@@ -310,6 +314,67 @@ implementation matches spec
 
 Initial raw version. Released []()
 
+# Upgradability and Compatibility
+## Compatibility with Waku v1
+
+Waku v2 and Waku v1 are different protocols all together. They use a different
+transport protocol underneath; Waku v1 is devp2p RLPx based while Waku v2 uses
+libp2p. The protocols themselves also differ as does their data format.
+Compatibility can be achieved only by using a bridge that not only talks both
+devp2p RLPx and libp2p, but that also transfers (partially) the content of a
+packet from one version to the other.
+
+### Bridge
+
+A bridge requires supporting both Waku versions:
+
+* Waku v1 - using devp2p RLPx protocol
+* Waku v2 - using libp2p protocols
+
+Packets received on the Waku v1 network SHOULD be published just once on the
+Waku v2 network. More specifically, the bridge SHOULD publish
+this through the Waku Relay (PubSub domain).
+
+Publishing such packet will require the creation of a new `Message` with a
```

Comment:
I suppose dealing with the contents of `data` field is out of scope of this PR and we can deal with it here https://github.com/vacp2p/specs/issues/181 there's a question of Waku v2 receiving RLP envelopes here, but not much that can be done about it at this level I suppose.

Possibly some indication that it came from Waku v1 network? That way clients using Waku v2 can decide to drop support for RLPx in future when it doesn't feel like supporting that data content anymore. WDYT? Could be indicated in the contentTopic field perhaps, such as `wakuv1/contenttopic` (similar to eth2 namespaces). Or a separate field.

Reply:
The contents of data is indeed something to answer in #181.

But the idea is not to pass any RLP encoded data to the Waku v2 network. It is true that the Waku v1 `Envelope` type gets RLP encoded. But here it gets decoded and then we use the specific `data` and `payload` field an place it in a new (protobuf encoded) `WakuMessage`.

Of course of there is any RLP encoded data within the `data` field, that is up to the application.

Reply:
Currently the `data` field is part of the protocol (though optional as you note): https://specs.vac.dev/specs/waku/envelope-data-format.html - I suppose it can be read in a custom fashion, but it isn't a straightforward protobuf. Though the payload _within_ the `data` field is.

Anyway, for the other issue